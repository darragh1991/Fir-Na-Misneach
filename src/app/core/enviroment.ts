export function developmentInitializer() {
    console.log('Development environment initialized');
}

export function productionInitializer() {
    console.log('Production environment initialized');
}
