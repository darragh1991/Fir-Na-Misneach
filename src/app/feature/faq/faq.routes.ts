import { Routes } from "@angular/router";

export default <Routes> [
  {
    path: '',
    providers: [],
    children: [
      {
        path: '',
        loadComponent: () => import('./components/faq.component').then(m => m.FaqComponent)
      }
    ]
   }
]
