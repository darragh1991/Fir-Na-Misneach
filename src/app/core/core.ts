import {
  Routes,
  provideRouter,
  withComponentInputBinding,
  withEnabledBlockingInitialNavigation,
  withInMemoryScrolling,
  withRouterConfig
} from '@angular/router';
import { ENVIRONMENT_INITIALIZER, InjectionToken } from '@angular/core';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideHttpClient } from '@angular/common/http';
import { isDevMode } from '@angular/core';
import { developmentInitializer, productionInitializer } from './enviroment';

export interface CoreOptions {
  routes: Routes;
}

export function provideCore({ routes }: CoreOptions) {
  return [
    provideAnimationsAsync(),
    provideHttpClient(),
    provideRouter(
      routes,
      withRouterConfig({ onSameUrlNavigation: 'reload' }),
      withComponentInputBinding(),
      withEnabledBlockingInitialNavigation(),
      withInMemoryScrolling({
        anchorScrolling: 'enabled',
        scrollPositionRestoration: 'top'
      }),
    ),
    {
      provide: ENVIRONMENT_INITIALIZER,
      useValue: () => isDevMode() ? developmentInitializer() : productionInitializer(),
      multi: true
    }
  ];
}
