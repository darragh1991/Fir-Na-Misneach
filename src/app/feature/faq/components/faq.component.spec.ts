import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FaqComponent } from './faq.component';
import { provideHttpClient } from '@angular/common/http';

describe('FaqComponent', () => {
  let fixture: ComponentFixture<FaqComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [FaqComponent],
      providers: [
        provideHttpClient()
      ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(FaqComponent);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });
});
