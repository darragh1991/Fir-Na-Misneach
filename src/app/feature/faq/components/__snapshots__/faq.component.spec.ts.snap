// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqComponent should create 1`] = `
<app-faq
  faqService$={[Function Observable]}
>
  <!--container-->
</app-faq>
`;
